messages/src/main/java/no/unit/nva/publication/messages/create/NewCreateMessageHandler.java

```diff
@@ -87,14 +88,63 @@ private TicketEntry fetchTicketForUser(RequestInfo requestInfo, SortableIdentifi
                                            UserInstance user)
         throws ApiGatewayException {
         return userIsElevatedUser(requestInfo)
-                   ? fetchTicketForElevatedUser(ticketIdentifier, user)
-                   : fetchTicketForPublicationOwner(ticketIdentifier, user);
+                ? fetchTicketAndValidateAccessRightsForElevatedUser(requestInfo, ticketIdentifier, user)
+                : fetchTicketForPublicationOwner(ticketIdentifier, user);
+    }
+
+    private static boolean userIsElevatedUser(RequestInfo requestInfo) {
+        return userIsAuthorizedToApproveDoiRequest(requestInfo)
+                || userIsAuthorizedToApprovePublishingRequest(requestInfo);
+    }
+
+    private TicketEntry fetchTicketAndValidateAccessRightsForElevatedUser(RequestInfo requestInfo,
+                                                                          SortableIdentifier ticketIdentifier,
+                                                                          UserInstance user)
+            throws ApiGatewayException {
+        var ticketEntry = fetchTicketForElevatedUser(ticketIdentifier, user);
+        validateAccessRightsForTicketType(requestInfo, ticketEntry.getClass());
+        return ticketEntry;
     }
 
     private TicketEntry fetchTicketForPublicationOwner(SortableIdentifier ticketIdentifier, UserInstance user)
-        throws ApiGatewayException {
+            throws ApiGatewayException {
         return attempt(() -> ticketService.fetchTicket(user, ticketIdentifier))
-                   .orElseThrow(fail -> handleFetchingTicketForUserError(fail.getException()));
+                .orElseThrow(fail -> handleFetchingTicketForUserError(fail.getException()));
+    }
+
+    private void validateAccessRightsForTicketType(RequestInfo requestInfo, Class<? extends TicketEntry> ticketType)
+            throws ApiGatewayException {
+        if (userIsNotAuthorizedToCreateMessageForDoiRequest(requestInfo, ticketType)
+                || userIsNotAuthorizedToCreateMessageForPublishingRequestCase(requestInfo, ticketType)) {
+            throw new ForbiddenException();
+        }
+    }
+
+    private boolean userIsNotAuthorizedToCreateMessageForPublishingRequestCase(
+            RequestInfo requestInfo, Class<? extends TicketEntry> ticketType) {
+        return ticketType.equals(PublishingRequestCase.class)
+                && userIsNotAuthorizedToApprovePublishingRequest(requestInfo);
+    }
+
+    private boolean userIsNotAuthorizedToCreateMessageForDoiRequest(
+            RequestInfo requestInfo, Class<? extends TicketEntry> ticketType) {
+        return ticketType.equals(DoiRequest.class) && userIsNotAuthorizedToApproveDoiRequest(requestInfo);
+    }
+
+    private boolean userIsNotAuthorizedToApprovePublishingRequest(RequestInfo requestInfo) {
```

Comment:
Gåt det an å bruke noe a lá Map<Class<TicketEntry>, String> ticketTypeAndCorrespondingAccessRight = Map.of(DoiRequest.class, ACCESS_RIGHT_APPROVE_DOI_REQUEST...

requestInfo.userIsAuthorized(ticketTypeAndCorrespondingAccessRight.get(ticketType);

Da sparer du deg for noen if-statements